js/content/community.js

```diff
@@ -3451,8 +3451,79 @@ let GuidesPageClass = (function(){
     return GuidesPageClass;
 })();
 
+let WorkshopPageClass = (function(){
 
-class WorkshopPageClass {
+    function WorkshopPageClass() {
+        this.loadLastState();
+        this.initAjaxBrowse();
+    }
+
+    WorkshopPageClass.prototype.loadLastState = function() {
+        let url = new URL(window.location.href);
+
+        if (url.searchParams && url.searchParams.has("browsesort")) {
+            LocalStorage.set("workshop_state", url.search);
+            return;
+        }
+
+        let search = LocalStorage.get("workshop_state");
+        url = new URL("https://steamcommunity.com/workshop/" + search);
+        let query = url.searchParams.get("browsesort");
+        this.changeTab(query);
+    };
+
+    WorkshopPageClass.prototype.initAjaxBrowse = function() {
+        ExtensionLayer.runInPageContext(function() {
+            $J(".browseOption").get().forEach(node => node.onclick = () => false);
+        });
+
+        document.querySelectorAll(".browseOption").forEach(tab => {
+            tab.addEventListener("click", () => {
+                let a = tab.querySelector("a[href]");
+                let url = new URL("https://steamcommunity.com/workshop/" + a.href);
+                let query = url.searchParams.get("browsesort");
+                LocalStorage.set("workshop_state", url.search);
+                window.history.pushState(null, null, url.search);
+                this.changeTab(query);
+            });
+        });
+    };
+
+    WorkshopPageClass.prototype.changeTab = async function(query, start=0, count=8) {
```

Comment:
```suggestion
    WorkshopPageClass.prototype.changeTab = async function(query) {
```
I can't imagine ever using these parameters

Reply:
I am using it. It also makes it a more portable function.

Reply:
When would you ever want to use that?

Reply:
And also, you're obviously not using it, since I don't see anything of it in the code.
If you use this for testing only:
You also use debug logs for testing only, but you remove it when pushing

Reply:
It was not used for testing. It was added to increase portability, usability and somewhat maintainability.

Reply:
I mean, I'd accept your argument if you would add these default parameters to a more universal function, but changing tabs in the small window of the workshop homepage is not what I consider universal

Reply:
Who knows, maybe in the future we add functionality to link to certain pages. This function would be able to ajax load them. I'm just saying, this function with parameters is fine, if not better.